announcement/models/announcement.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2022 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class AnnouncementLog(models.Model):
+    _name = "announcement.log"
+    _description = "Log user reads"
+    _order = "create_date desc"
+
+    announcement_id = fields.Many2one(comodel_name="announcement")
+
+
+class Announcement(models.Model):
+    _name = "announcement"
+    _description = "User announcements"
+    _order = "sequence asc, id"
+
+    active = fields.Boolean()
+    sequence = fields.Integer()
+    name = fields.Char(string="Title", required=True)
+    content = fields.Html()
+    announcement_type = fields.Selection(
+        selection=[
+            ("specific_users", "Specific users"),
+            ("user_group", "User groups"),
+        ],
+        default="specific_users",
+        required=True,
+    )
+    specific_user_ids = fields.Many2many(
+        comodel_name="res.users",
+        domain=[("share", "=", False)],
+        inverse="_inverse_specific_user_ids",
+    )
+    user_group_ids = fields.Many2many(comodel_name="res.groups",)
+    notification_date = fields.Datetime()
+    read_announcement_count = fields.Integer(
+        compute="_compute_read_announcement_count",
+    )
+
+    def _inverse_specific_user_ids(self):
+        for announcement in self:
+            for user in announcement.specific_user_ids.filtered(
+                lambda x: announcement
+                not in (x.read_announcement_ids + x.pendant_announcement_ids)
+            ):
+                user.pendant_announcement_ids |= announcement
+
+    @api.onchange("announcement_type")
+    def _onchange_announcement_type(self):
+        if self.announcement_type == "specific_users":
+            self.user_group_ids = False
+        elif self.announcement_type == "user_group":
+            self.specific_user_ids = False
+
+    def _compute_read_announcement_count(self):
+        for announcement in self:
+            announcement.read_announcement_count = self.env[
+                "announcement.log"
+            ].search_count([("announcement_id", "=", announcement.id)])
```

Comment:
```suggestion
        logs = self.env["announcement.log"].read_group(
            [("announcement_id", "in", self.ids)],
            ["announcement_id"],
            ["announcement_id"],
        )
        result = {
            data["announcement_id"][0]: (data["announcement_id_count"]) for data in logs
        }
        for announcement in self:
            announcement.read_announcement_count = result.get(announcement.id, 0)
```

Reply:
Thanks :) I was going to use this field only in the form, but thinking twice I get your code and I'll be showing the count as well in the tree :smile: :+1: